common/pb/event_move_card.proto

```diff
@@ -16,4 +16,5 @@ message Event_MoveCard {
     optional sint32 y = 9 [default = -1];
     optional sint32 new_card_id = 10 [default = -1];
     optional bool face_down = 11;
+    repeated sint32 swap_indexes = 12;
```

Comment:
this will not be backwards compatible, old clients will be unable to represent this information, resulting in the shuffle not happening at all for them, possibly causing them to become desynchronized and having cards swap places depending on the viewer (moving cards between zones fixes such desynchronizations)

Reply:
hrm, could we maybe use position? it's signed so we could associate this with a large negative number that would otherwise not be possible to get, then either encode the indexes in the int or have it be a random number that's then used as a rand seed. i don't know if that fully fixes the backwards compatibility issue though since you'd still need the new client to translate.

Reply:
that's just as much not backwards compatible (and more confusing), it's not about not adding new fields it's about the behavior.
my proposal is still to delete the set of old cards and create new ones in a random order, that way it's not possible for clients to track which card is which as they'll be given new ids.
alternatively we could switch the ids of each of the cards without telling the clients which is which until flipped, giving like a single new event telling them which cards are affected, new clients could forget which id is associated with each card name then, while old clients would still see the old card until it's flipped over I guess.
the part that's most important to fix is that clients are told about the outcome of the randomisation, clients shouldn't be given what would be unknown information at all, even if the client we provide would hide it.